cmd/agent/app/reporter/flags.go

```diff
@@ -48,19 +46,24 @@ type Options struct {
 
 // AddFlags adds flags for Options.
 func AddFlags(flags *flag.FlagSet) {
-	flags.String(reporterType, string(GRPC), fmt.Sprintf("Reporter type to use e.g. %s, %s[%s]", string(GRPC), string(TCHANNEL), "NOTE: Deprecated since 1.16"))
-	if !setupcontext.IsAllInOne() {
-		flags.String(agentTagsDeprecated, "", "(deprecated) see --"+agentTags)
-		flags.String(agentTags, "", "One or more tags to be added to the Process tags of all spans passing through this agent. Ex: key1=value1,key2=${envVar:defaultValue}")
+	AddFlagsWithCustomTypes([]string{})(flags)
+}
+
+// AddFlagsWithCustomTypes adds types to use message.
+func AddFlagsWithCustomTypes(types []string) func(*flag.FlagSet) {
```

Comment:
Why do we need `types`? 

Also - if `GRPC` is the only supported reporter, we can deprecate the flag so that we can remove it later

Reply:
I made it backwars compatible, so in Uber's custom build the type can be added back.

> Also - if GRPC is the only supported reporter, we can deprecate the flag so that we can remove it later

I would leave it for now as it is. It gives us more flexibility to extend.

Reply:
I have added a better description to the signature